### PR TITLE
Contributing to docs: assembly/module file metadata improvements

### PR DESCRIPTION
**Pull request by apinnick**
Moved guidelines in the assembly and module file metadata sections to improve the organization.

Version(s): NA

Issue: None

Link to docs preview: None. On the **Files changed** tab, click **... > View file**

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: contributing_to_docs/doc_guidelines.adoc [hunk 1]
```diff
@@ -72,8 +72,8 @@ toc::[]
...
-* Do not use "Optional:" to prefix a heading title. According to IBM Style, this practice is reserved for optional steps and code callouts. With content reuse and modular documentation, you cannot be certain that all future use of a module will remain optional. For more information, see link:https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps["Indicating optional and conditional steps (Requires IBM Style login)"].
+* Do not use backticks or other markup in assembly headings. You can use backticks or other markup in the title for a block, such as a code block `.Example` or a table `.Description` title.
+* Do not use "Optional:" to prefix a heading title. According to IBM Style, this practice is reserved for optional steps and code callouts. With content reuse and modular documentation, you cannot be certain that all future use of an assembly will remain optional. For more information, see link:https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps["Indicating optional and conditional steps (Requires IBM Style login)"].
```
[the remark this answers is not in the document]
Comment (apinnick) [reply contributing_to_docs/doc_guidelines.adoc]: I will create a Vale rule to allow "IBM Style".

File: contributing_to_docs/doc_guidelines.adoc [hunk 2]
```diff
@@ -33,20 +33,30 @@ In the Pulsar editor, you can use `Ctrl`+`J` to undo hard wrapping on a paragrap
 
 [id="assembly-file-metadata"]
 == Assembly file metadata
-Every assembly file should contain the following metadata at the top, with no line spacing in between, except where noted:
+Every assembly file should contain the following metadata at the top, with no blank lines, except where noted:
```
Comment (ShaunaDiaz) [inline contributing_to_docs/doc_guidelines.adoc:36]:
```suggestion
Every assembly file must contain the following metadata at the top, with no blank lines, except where noted:
```

File: contributing_to_docs/doc_guidelines.adoc [hunk 3]
```diff
@@ -33,20 +33,30 @@ In the Pulsar editor, you can use `Ctrl`+`J` to undo hard wrapping on a paragrap
...
 toc::[]                                                         <7>
 ----
 <1> The content type for the file. For assemblies, always use `:_mod-docs-content-type: ASSEMBLY`. Place this attribute before the anchor ID or, if present, the conditional that contains the anchor ID.
 <2> A unique (within OpenShift docs) anchor ID for this assembly. Use lowercase and hyphens, for example: `cli-developer-commands`.
 <3> Human readable title (notice the `=` top-level header).
++
+[NOTE]
+====
+* Do not use backticks or other markup in assembly headings. You can use backticks or other markup in the title for a block, such as a code block `.Example` or a table `.Description` title.
+* Do not use "Optional:" to prefix an assembly title. According to IBM Style, this practice is reserved for optional steps and code callouts. With content reuse and modular documentation, you cannot be certain that all future use of an assembly will remain optional. For more information, see link:https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps["Indicating optional and conditional steps (Requires IBM Style login)"].
```
Comment (ShaunaDiaz) [inline contributing_to_docs/doc_guidelines.adoc:54]:
```suggestion
* Do not use "Optional:" to prefix an assembly title. According to the IBM Style Guide, using "Optional:" is reserved for optional steps and code callouts. 

* With content reuse, you cannot be certain that any assembly will be an optional topic in the future. For more information, see link:https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps["Indicating optional and conditional steps (Requires IBM Style login)"].
```

File: contributing_to_docs/doc_guidelines.adoc [hunk 4]
```diff
@@ -33,20 +33,30 @@ In the Pulsar editor, you can use `Ctrl`+`J` to undo hard wrapping on a paragrap
...
+* The assembly title, which is the first line of the document, is the only level 1 ( `=` ) title.
+Section headers within the assembly must be level 2 ( `==` ) or lower. When you include modules, you must add
```
Comment (ShaunaDiaz) [inline contributing_to_docs/doc_guidelines.adoc:56]:
```suggestion
Section headers within the assembly must be level 2 ( `==` ) or lower. When you include modules, you must add leveloffsets in the include statements. You can manually add more level 2 or lower section headers in the assembly.
```

Comment (ShaunaDiaz) [reply contributing_to_docs/doc_guidelines.adoc:56]: Funky line break in there

File: contributing_to_docs/doc_guidelines.adoc [hunk 5]
```diff
@@ -101,26 +99,14 @@ Every module should be placed in the modules folder and should contain the follo
 
 <1> List of assemblies in which this module is included.
 <2> The content type for the file. Replace `<TYPE>` with the actual type of the module, `CONCEPT`, `REFERENCE`, or `PROCEDURE`. Place this attribute before the anchor ID or, if present, the conditional that contains the anchor ID.
-<3> A module anchor with {context} that must be lowercase and must match the module's file name.
-<4> Human readable title. To ensure consistency in the results of the
-leveloffset values in include statements, you must use a level one heading
+<3> The anchor ID with `_{context}` must be lowercase and match the module's file name.
+<4> Human readable title. To ensure consistency in the results of the leveloffset values in include statements, you must use a level one heading
```
Comment (ShaunaDiaz) [inline contributing_to_docs/doc_guidelines.adoc:103]:
```suggestion
<4> Human readable title. To ensure consistency in the results of the leveloffset values in include statements, you must use a level 1 heading
```

File: contributing_to_docs/doc_guidelines.adoc [hunk 6]
```diff
@@ -101,26 +99,14 @@ Every module should be placed in the modules folder and should contain the follo
...
-[id="cli-basic-commands_{context}"]
-= Basic CLI commands
-----
-
++
 [NOTE]
 ====
-Do not use backticks or other markup in assembly or module headings. You can use backticks or other markup in the title for a block, such as a code block `.Example` or a table `.Description` title.
+* Do not use backticks or other markup in module headings. You can use backticks or other markup in the title for a block, such as a code block `.Example` or a table `.Description` title.
+* Do not use "Optional:" to prefix a module title. According to IBM Style, this practice is reserved for optional steps and code callouts. With content reuse and modular documentation, you cannot be certain that all future use of a module will remain optional. For more information, see link:https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps["Indicating optional and conditional steps (Requires IBM Style login)"].
```
Comment (ShaunaDiaz) [inline contributing_to_docs/doc_guidelines.adoc:109]: as suggested above

**Comment by ShaunaDiaz** [comment]
Just a couple of suggestions if we're here anyway. Nice contribution--thanks1

**Pull request merged**